wordpress-mobile/WordPress-iOS
# iPad/iPhone: Normalize header and footer views

Refs #4128, #5536

This PR cleans up a few inconsistencies across WPiOS on particularly iPad, but also iPhone. There were several instances of forcing/hacking header and footer heights instead of letting iOS provide the default header/footer height sizing within tableViews.

There were also instances of forcing header padding on iPad to satiate the 600px design, which in-turn created undesired header spacing when viewing the same screens within split-view multitasking.

To try and make some sense of it all, here's a summary:

- `BlogListViewController`: Normalized the header usage and padding.
- `SupportViewController`: Normalized the header usage and padding.
- `Reader`: Removes forced extra padding on iPad, now matching iPhone.
- `BlogSelectorViewController`: Normalized the header usage and padding.
- `PostCategoriesViewController`: Using grouped table styling for normalized padding.
- `NotificationsViewController`: Removed the header margin hack to instead default to no top margin/inset on iPad, for now.
- `NotificationDetailsViewController`: Conditional header margin when iPad is in full-screen and not split-view.
- `CommentsViewController`: Conditional header margin when iPad is in full-screen and not split-view.
- `CommentViewController`: Conditional header margin when iPad is in full-screen and not split-view.
- `WPGUIConstants`: Removed `WPTableHeaderPadFrame` and `WPTableFooterPadFrame`.

Note: this isn't a perfect PR as there are still inconsistencies resulting from the use of `WPTableViewSectionHeaderFooterView `, which will be nuked soon, but can only be nuked after nuking `WPTableViewCell` and its margin hacks. See (#5576) ☣️💥☣️💥☣️

To test:

1. Open each controller mentioned above and ensure the padding is as expected on iPad and iPhone.
2. Open the `CommentsViewController` and `CommentViewController` within a split-view iPad multitasking state to ensure no extra padding is present in the header.
3. Open the `NotificationDetailsViewController` within a split-view iPad multitasking state to ensure no extra padding is present in the header.
4. Don't 🔪 me in the process.

Needs review: @jleandroperez
